src/gui/components/record/conflict/conflictdroplist.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2021, 2022 Macquarie University
+ *
+ * Licensed under the Apache License Version 2.0 (the, "License");
+ * you may not use, this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND either express or implied.
+ * See, the License, for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Filename: conflictdroplist.tsx
+ * Description:
+ *   File about Revision Droplist, user can choose revision to resolve or edit from droplist
+ */
+import * as React from 'react';
+import {useState} from 'react';
+import Box from '@mui/material/Box';
+import Grid from '@mui/material/Grid';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import FormHelperText from '@mui/material/FormHelperText';
+import Select, {SelectChangeEvent} from '@mui/material/Select';
+import {Typography} from '@mui/material';
+import {BasicDiaglog} from './conflictDialog';
+type ConflictDropSelectprops = {
+  label: string;
+  headerlist: Array<string>;
+  revision: string;
+  index: number;
+  setRevision: any;
+  disablerevision: string;
+  islabel?: boolean;
+};
+
+function RevisionDropList(props: ConflictDropSelectprops) {
+  const {
+    label,
+    headerlist,
+    index,
+    setRevision,
+    disablerevision,
+    islabel,
+  } = props;
+  const [value, setValue] = useState(props.revision);
+  const [open, setOpen] = React.useState(false);
+  const [temvalue, settemvalue] = useState(props.revision);
+
+  const handleChange = (event: SelectChangeEvent) => {
+    if (props.revision !== '') {
+      settemvalue(event.target.value as string);
+      setOpen(true);
+    } else {
+      setRevision(event.target.value as string, index);
+      setValue(event.target.value as string);
+    }
+  };
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+  const handleClose = () => {
+    setOpen(false);
+    settemvalue(value);
+  };
+
+  const handleConfirm = () => {
+    setOpen(false);
+    setRevision(temvalue, index);
+    setValue(temvalue);
+  };
+
+  return (
+    <>
+      <Grid>
+        <BasicDiaglog
+          handleClose={handleClose}
+          handleOpen={handleOpen}
+          handleConfirm={handleConfirm}
+          content={
+            'If you proceed to change Conflict revision, all changes made here will be lost'
```

Comment:
If you change the conflict revision, any changes made here will be lost. Proceed?